unified-runtime/source/adapters/level_zero/v2/event.cpp

```diff
@@ -232,6 +237,14 @@ ur_result_t urEventRelease(ur_event_handle_t hEvent) try {
 ur_result_t urEventWait(uint32_t numEvents,
                         const ur_event_handle_t *phEventWaitList) try {
   for (uint32_t i = 0; i < numEvents; ++i) {
+    if (!phEventWaitList[i]->getIsEventInUse()) {
+      // TODO: This is a workaround for the underlying inconsistency
```

Comment:
Repeating comment from the original PR: can't we manually signal the events to put them in a proper state?

Reply:
No, it's not possible to signal or reset counter-based events from host. They also need to be previously used as part of another command append before they are usable.